File: app/src/main/java/com/example/angelo/todo/services/TaskService.java
package com.example.angelo.todo.services;

import android.util.Log;

import com.example.angelo.todo.database.TodoDatabaseHelper;
import com.example.angelo.todo.models.Task;
import com.example.angelo.todo.repositories.TaskRepository;

import java.util.ArrayList;

public class TaskService {

    private TaskRepository taskRepository;

    public TaskService(TodoDatabaseHelper database) {
        this.taskRepository = new TaskRepository(database);
    }

    public Task insert(String name) {
        Task task = new Task(name);

        long _id = this.taskRepository.insert(task);
        Long _idLong = new Long(_id);

        task.setId(_idLong.intValue());

        return task;
    }

    public void delete(long _id) {
        Long _idLong = new Long(_id);

        Task task = new Task("");
        task.setId(_idLong.intValue());

        this.taskRepository.delete(task);
    }

    public void update(long _id, String name) {
        Long _idLong = new Long(_id);

        Task task = new Task(name);
        task.setId(_idLong.intValue());

        this.taskRepository.update(task);
    }

    public ArrayList<Task> list() {
        return this.taskRepository.list();
    }

}

File: app/src/main/java/com/example/angelo/todo/services/TaskService.kt
package com.example.angelo.todo.services

import com.example.angelo.todo.database.DatabaseHelper
import com.example.angelo.todo.models.Task
import com.example.angelo.todo.repositories.TaskRepository
import com.example.angelo.todo.mapper.TaskMapper

import java.util.ArrayList

class TaskService(database: DatabaseHelper) {

    private val taskRepository: TaskRepository = TaskRepository(database)

    fun insert(name: String): Task {
        return taskRepository.insert(TaskMapper.toEntity(name))
    }

    fun delete(_id: Long) {
        this.taskRepository.delete(TaskMapper.toEntity(java.lang.Long.toString(_id), ""))
    }

    fun update(_id: Long, name: String): Task {
        return taskRepository.update(TaskMapper.toEntity(java.lang.Long.toString(_id), name))
    }

    fun list(): ArrayList<Task> {
        return this.taskRepository.list()
    }
}
